frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -1543,4 +1561,68 @@ private void setDefaultRequestAttributes(RenderRequest request) {
                 p -> nullToEmptyString(p.getState()));
         return isAscending ? comparator : comparator.reversed();
     }
+
+	private Object[] getCyclicProjectPresenceAndLastProjectInCycle(Project project, ProjectService.Iface client,
+            User user, Map<String, String> linkedProjectPath) throws TException {
+
+        Iterator<String> linkedProjectIterator = project.getLinkedProjects().keySet().iterator();
+
+        while (linkedProjectIterator.hasNext()) {
+            String linkedProjectId = linkedProjectIterator.next();
+            Project linkedProject = client.getProjectByIdForEdit(linkedProjectId, user);
+            String projectName = linkedProject.getName();
+            String projectVersion = linkedProject.getVersion();
+            String projFullName = getFullProjectName(projectName, projectVersion);
+            if (linkedProjectPath.containsKey(linkedProjectId)) {
+                return new Object[] { Boolean.TRUE, projFullName };
+            }
+
+            linkedProjectPath.put(linkedProjectId, projFullName);
+
+            Object[] cyclicLinkedProjectPresenceAndLastProjectInCycle = getCyclicProjectPresenceAndLastProjectInCycle(linkedProject,
+                   client, user, linkedProjectPath);
+            boolean isCyclicLinkedProjectPresent = (Boolean) cyclicLinkedProjectPresenceAndLastProjectInCycle[0];
+
+            if (isCyclicLinkedProjectPresent) {
+                return cyclicLinkedProjectPresenceAndLastProjectInCycle;
+            }
+            linkedProjectPath.remove(linkedProjectId);
+        }
+
+        return new Object[] { Boolean.FALSE, null };
+    }
+
+	private String getFullProjectName(String projectName, String projectVersion) {
+        StringBuilder fullProjectName = new StringBuilder(projectName);
+
+        if (projectVersion != null && !projectVersion.isEmpty()) {
+            fullProjectName.append(" (").append(projectVersion).append(")");
+        }
+
+        return fullProjectName.toString();
+    }
+
+    private boolean isCyclicLinkedProjectPresentAndSetParams(String id,Project project,ProjectService.Iface client, User user, ActionRequest request, ActionResponse response) throws TException
+    {
+        Map<String,String> linkedProjectPath=new LinkedHashMap<>();
+        String firstProjFullName = getFullProjectName(project.getName(),project.getVersion());
+        linkedProjectPath.put(id,firstProjFullName);
+        Object[] cyclicLinkedProjectPresenceAndLastProjectInCycle = getCyclicProjectPresenceAndLastProjectInCycle(project, client, user, linkedProjectPath);
+		
+        boolean isCyclicLinkedProjectPresent = (Boolean) cyclicLinkedProjectPresenceAndLastProjectInCycle[0];
+		if (isCyclicLinkedProjectPresent) {
+            Collection<String> values = linkedProjectPath.values();
+            String lasttProjInCycle = (String) cyclicLinkedProjectPresenceAndLastProjectInCycle[1];
+            String cyclicHierarchy = String.join(SEPARATOR, values);
```

Comment:
Convert `linkedProjectPath.values()` to array and then use `String.join()`.

Reply:
Converted linkedProjectPath.values() to array before using String.join() as suggested